src/PerfView/StackViewer/StackWindow.xaml.cs

```diff
@@ -1586,12 +1586,23 @@ private void DoSetTimeRange(object sender, ExecutedRoutedEventArgs e)
             var cells = SelectedCells();
             if (cells != null)
             {
-                if (cells.Count == 2)
+                if (cells.Count == 2) // First and Last column cells
                 {
                     StartTextBox.Text = GetCellStringValue(cells[0]);
                     EndTextBox.Text = GetCellStringValue(cells[1]);
                     Update();
                 }
+                else if (cells.Count == 1) // single cell from a row
```

Comment:
I would prefer that this look just like the 2 element case as much as possible.  

Indeed by preferred code is  to fold it into the case above 

```
if (cells.Count == 2 || cells.Count == 1) {
    StartTextBox.Text = GetCellStringValue(cells[0]);
    if (cells.Count == 1)
        EndTextBox.Text = GetCellStringValue(cells[0]);
    else 
        EndTextBox.Text =  StartTextBox.Text ;
    Update();
}
```

Reply:
in case cells.Count == 1 we have single cell selected. We don't know if it's a Start/Last column. It can be something else (like name), in that case your proposal would fail.